### ISSUE TITLE
Exception from cobertura plugin during maven site generation

### ISSUE THREAD

**Issue by romani** [checkstyle/checkstyle#805]
"mvn clean site"

that is caused by some latest updates, I do not remember that ERROR few day ago.

```
[ERROR] Mar 15, 2015 8:52:57 PM net.sourceforge.cobertura.coveragedata.CoverageDataFileHandler loadCoverageData
INFO: Cobertura: Loaded information on 353 classes.
net.sourceforge.cobertura.javancss.parser.ParseException: Encountered " "final" "final "" at line 277, column 14.
Was expecting one of:
    "assert" ...
    "boolean" ...
    "byte" ...
    "char" ...
    "double" ...
    "enum" ...
    "false" ...
    "float" ...
    "int" ...
    "long" ...
    "new" ...
    "null" ...
    "short" ...
    "super" ...
    "this" ...
    "true" ...
    "void" ...
    <INTEGER_LITERAL> ...
    <FLOATING_POINT_LITERAL> ...
    <CHARACTER_LITERAL> ...
    <STRING_LITERAL> ...
    <IDENTIFIER> ...
    "(" ...
    ")" ...
    "!" ...
    "~" ...
    "++" ...
    "--" ...
    "+" ...
    "-" ...
    
	at net.sourceforge.cobertura.javancss.parser.JavaParser.generateParseException(JavaParser.java:11869)
	at net.sourceforge.cobertura.javancss.parser.JavaParser.jj_consume_token(JavaParser.java:11730)
	at net.sourceforge.cobertura.javancss.parser.JavaParser.TryWithResources(JavaParser.java:4618)
	at net.sourceforge.cobertura.javancss.parser.JavaParser.TryStatement(JavaParser.java:4533)
	at net.sourceforge.cobertura.javancss.parser.JavaParser.Statement(JavaParser.java:3754)
	at net.sourceforge.cobertura.javancss.parser.JavaParser.BlockStatement(JavaParser.java:3906)
	at net.sourceforge.cobertura.javancss.parser.JavaParser.Block(JavaParser.java:3855)
	at net.sourceforge.cobertura.javancss.parser.JavaParser.MethodDeclaration(JavaParser.java:1933)
	at net.sourceforge.cobertura.javancss.parser.JavaParser.ClassBodyDeclaration(JavaParser.java:1012)
	at net.sourceforge.cobertura.javancss.parser.JavaParser.ClassBody(JavaParser.java:874)
	at net.sourceforge.cobertura.javancss.parser.JavaParser.UnmodifiedClassDeclaration(JavaParser.java:801)
	at net.sourceforge.cobertura.javancss.parser.JavaParser.ClassDeclaration(JavaParser.java:706)
	at net.sourceforge.cobertura.javancss.parser.JavaParser.TypeDeclaration(JavaParser.java:555)
	at net.sourceforge.cobertura.javancss.parser.JavaParser.CompilationUnit(JavaParser.java:303)
	at net.sourceforge.cobertura.javancss.parser.JavaParser.parse(JavaParser.java:100)
	at net.sourceforge.cobertura.javancss.Javancss._measureSource(Javancss.java:192)
	at net.sourceforge.cobertura.javancss.Javancss._measureRoot(Javancss.java:268)
	at net.sourceforge.cobertura.javancss.Javancss.<init>(Javancss.java:346)
	at net.sourceforge.cobertura.reporting.ComplexityCalculator.getAccumlatedCCNForSource(ComplexityCalculator.java:105)
	at net.sourceforge.cobertura.reporting.ComplexityCalculator.getAccumlatedCCNForSingleFile(ComplexityCalculator.java:143)
	at net.sourceforge.cobertura.reporting.ComplexityCalculator.getCCNForSourceFileNameInternal(ComplexityCalculator.java:234)
	at net.sourceforge.cobertura.reporting.ComplexityCalculator.getCCNForPackageInternal(ComplexityCalculator.java:201)
	at net.sourceforge.cobertura.reporting.ComplexityCalculator.getCCNForProject(ComplexityCalculator.java:167)
	at net.sourceforge.cobertura.reporting.html.HTMLReport.generateTableRowForTotal(HTMLReport.java:603)
	at net.sourceforge.cobertura.reporting.html.HTMLReport.generateOverview(HTMLReport.java:287)
	at net.sourceforge.cobertura.reporting.html.HTMLReport.generateOverviews(HTMLReport.java:228)
	at net.sourceforge.cobertura.reporting.html.HTMLReport.<init>(HTMLReport.java:71)
	at net.sourceforge.cobertura.reporting.Main.parseArguments(Main.java:103)
	at net.sourceforge.cobertura.reporting.Main.main(Main.java:169)
```

**Comment by romani** [comment]
@mkordas , please take a look, we need to resolve that till 6.5 release.

**Comment by mkordas** [comment]
OK, please assign me.

**Comment by mkordas** [comment]
Problem is caused by `final` used by try-with-resources used in [Main.java](../blob/master/src/main/java/com/puppycrawl/tools/checkstyle/Main.java#L275). It should have been fixed in cobertura/cobertura#87 and released with Cobertura 2.1.1 and cobertura-maven-plugin 2.7. I cannot update them due to the other build errors, so for now I'll just remove `final`.
